File: features/profile/profileService.ts
// Profile service
// TODO: Implement using @/lib/supabase

export async function getProfile(userId: string) {
  // TODO: Implement
  return null;
}

export async function updateProfile(
  userId: string,
  data: Partial<{ displayName: string; avatarUrl: string; bio: string }>
) {
  // TODO: Implement
  return { success: false };
}

File: Gym/Features/Profile/ProfileService.swift
import Foundation
import Supabase

enum ProfileService {
    private static var client: Supabase.SupabaseClient {
        SupabaseClientProvider.shared
    }

    static func getProfile(userId: String) async throws -> Profile? {
        // TODO: Implement
        nil
    }

    static func updateProfile(
        userId: String,
        displayName: String? = nil,
        avatarUrl: String? = nil,
        bio: String? = nil
    ) async throws {
        // TODO: Implement
    }
}
